frontend/src/pages/Match/index.js

```diff
@@ -80,9 +80,9 @@ function MatchPage() {
   }, [difficulty, socket, userCtx]);
   useEffect(() => {
     if (room && navigate) {
-      room && navigate('/room', { state: { room } });
+      room && navigate('/room', { state: { room: room, difficulty: difficulty } });
     }
-  }, [room, navigate]);
+  }, [room, navigate, difficulty]);
```

Comment:
Similarly, have u checked by putting a console.log within this component to see if it keeps rendering infinitely? Remember to remove console.log after testing, thanks!

Reply:
Yes this does not render infinitely 
![image](https://user-images.githubusercontent.com/56862300/198841922-abee7e0f-03d1-4cc6-8d4c-80527802c669.png)